### PR TITLE
Add com.bktus.gpgfrontend

### PR DESCRIPTION
**Pull request by saturneric**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: com.bktus.gpgfrontend.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        sha256: 8e8c2fcc982f9ca67dcbb1d95e2dc746b1739a4668bc20b3a3c5be632edb34e4
+  - name: gpgme
+    buildsystem: simple
+    build-commands:
+      - ./autogen.sh
+      - ./configure --prefix=${FLATPAK_DEST} --enable-maintainer-mode && make -j$(nproc) && make install
+    sources:
+      - type: archive
+        url: https://gnupg.org/ftp/gcrypt/gpgme/gpgme-1.20.0.tar.bz2
+        sha256: 25a5785a5da356689001440926b94e967d02e13c49eb7743e35ef0cf22e42750
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:59]: These all should be in the runtime KDE 6.4 is based on Freedesktop runtime 22.08

- gettext: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/b2e749da20a562587c9b8bc605ce30a0650f6881/elements/platform.bst#L15
- libgpg-error: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/b2e749da20a562587c9b8bc605ce30a0650f6881/elements/platform.bst#L39
- libassuan: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/b2e749da20a562587c9b8bc605ce30a0650f6881/elements/platform.bst#L32
- gpgme: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/b2e749da20a562587c9b8bc605ce30a0650f6881/elements/platform.bst#L23

File: com.bktus.gpgfrontend.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  - name: pinentry
+    buildsystem: simple
+    build-commands:
+      - ./autogen.sh
+      - ./configure --prefix=${FLATPAK_DEST} --enable-maintainer-mode && make -j$(nproc) && make install
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:64]: This should use autotools buildsystem I think

File: com.bktus.gpgfrontend.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
+app-id: com.bktus.gpgfrontend
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:1]: Do you own/control the domain here `bktus.com`?

File: com.bktus.gpgfrontend.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+runtime: org.kde.Platform
+runtime-version: '6.4'
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:3]: This is EOL

File: com.bktus.gpgfrontend.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  - name: pinentry
+    buildsystem: simple
+    build-commands:
+      - ./autogen.sh
+      - ./configure --prefix=${FLATPAK_DEST} --enable-maintainer-mode && make -j$(nproc) && make install
+    sources:
+      - type: archive
+        url: https://gnupg.org/ftp/gcrypt/pinentry/pinentry-1.2.1.tar.bz2
+        sha256: 457a185e5a85238fb945a955dc6352ab962dc8b48720b62fc9fa48c7540a4067
+  - name: boost
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:69]: This is needed at build time only? The add a cleanup: '*'

File: com.bktus.gpgfrontend.yaml (hunk 2 above)
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:64]:
```suggestion
    buildsystem: autotools
    config-opts:
      - --enable-maintainer-mode
```

Should use autotools, rest is figured out by flatpak-builder

File: com.bktus.gpgfrontend.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      - ./bootstrap.sh --prefix=${FLATPAK_DEST} --with-libraries=date_time,system
+      - ./b2 variant=release link=static runtime-link=shared cxxflags="$CXXFLAGS" linkflags="$LDFLAGS" -j $(nproc) install
+    sources:
+      - type: archive
+        url: https://boostorg.jfrog.io/artifactory/main/release/1.78.0/source/boost_1_78_0.tar.bz2
+        sha256: 8681f175d4bdb26c52222665793eef08490d7758529330f98d3b29dd0735bccc
+  - name: libconfig++
+    buildsystem: simple
+    build-commands:
+      - ./configure --prefix=${FLATPAK_DEST} && make -j$(nproc) && make install
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:81]:
```suggestion

```

File: com.bktus.gpgfrontend.yaml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  - --socket=wayland
+  - --filesystem=host
+  - --device=dri
+  - --share=network
+
+cleanup:
+  - /lib/*.a
+  - /lib/*.la
+  - /lib32/*.a
+  - /lib32/*.la
```
[the remark this answers is not in the document]
Comment (bbhtt) [reply com.bktus.gpgfrontend.yaml:19]: Add cleanups:

- /include
- /lib/cmake
- /lib/pkgconfig

File: com.bktus.gpgfrontend.yaml (hunk 2 above)
Comment (bbhtt) [reply com.bktus.gpgfrontend.yaml:64]: Do you need the Qt or any other backends? What you did builds only the curses one. If so add `--enable-pinentry-qt`

Comment (bbhtt) [reply com.bktus.gpgfrontend.yaml:64]: Is `/usr/bin` hardcoded in your program?

```
[14:06:40.700] [T:24] [core] [ error  ] [GpgCoreInit.cpp:159] [init_gpgfrontend_core] -> core loaded custom gpgconf path is illegal: gpgconf (+2us)
[14:06:40.700] [T:24] [core] [warning ] [GlobalSettingStation.h:78] [LookupSettings] -> setting not found: network.auto_import_missing_key (+0us)
[14:06:40.700] [T:24] [core] [ error  ] [GpgCoreInit.cpp:172] [init_gpgfrontend_core] -> core loaded custom gpg key database is illegal:  (+35us)
QProcess::readAllStandardError: Called with MergedChannels
```

programs in the manifest goes into `/app`, pinentry is in `/app/bin` inside flatpak

File: com.bktus.gpgfrontend.yaml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+runtime: org.kde.Platform
+runtime-version: '6.4'
+sdk: org.kde.Sdk
+command: GpgFrontend
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --filesystem=host
```
[the remark this answers is not in the document]
Comment (bbhtt) [reply com.bktus.gpgfrontend.yaml:11]: File open/save dialogue seems to support the filechooser portal

File: com.bktus.gpgfrontend.yaml (hunk 3 above)
Comment (saturneric) [reply com.bktus.gpgfrontend.yaml:1]: yes

File: com.bktus.gpgfrontend.metainfo.xml
[the remark this answers is not in the document]
Comment (saturneric) [reply com.bktus.gpgfrontend.metainfo.xml:1]: It is now placed here.

https://github.com/saturneric/GpgFrontend/blob/dev/2.1.1/main/resource/appstream/com.bktus.gpgfrontend.appdata.xml

File: com.bktus.gpgfrontend.desktop
[the remark this answers is not in the document]
Comment (saturneric) [reply com.bktus.gpgfrontend.desktop:1]: It is now placed here.

https://github.com/saturneric/GpgFrontend/blob/dev/2.1.1/main/resource/appstream/com.bktus.gpgfrontend.desktop

File: com.bktus.gpgfrontend.metainfo.xml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    <p>For moew information visit https://gpgfrontend.bktus.com/</p>
+
+  </description>
+
+  <launchable type="desktop-id">com.bktus.gpgfrontend.desktop</launchable>
+
+  <screenshots>
+    <screenshot type="default">
+      <caption>Main Window</caption>
+      <image>https://raw.githubusercontent.com/saturneric/Blob/master/screenshots/keygen_ubuntu.png</image>
```
[the remark this answers is not in the document]
Comment (saturneric) [reply com.bktus.gpgfrontend.metainfo.xml]: Fixed.

File: com.bktus.gpgfrontend.metainfo.xml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    </screenshot>
+  </screenshots>
+
```
[the remark this answers is not in the document]
Comment (saturneric) [reply com.bktus.gpgfrontend.metainfo.xml]: Fixed

File: com.bktus.gpgfrontend.yaml (hunk 3 above)
Comment (saturneric) [reply com.bktus.gpgfrontend.yaml:1]: You can verify it by visiting https://www.gpgfrontend.bktus.com/

File: com.bktus.gpgfrontend.yaml [hunk 11]
```diff
@@ -0,0 +1,96 @@
+app-id: com.bktus.gpgfrontend
+runtime: org.kde.Platform
+runtime-version: '22.08'
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:3]: 22.08 of KDE doesn't exist. You probably want 6.5

File: com.bktus.gpgfrontend.yaml (hunk 1 above)
Comment (bbhtt) [reply com.bktus.gpgfrontend.yaml:59]: Drop these

File: com.bktus.gpgfrontend.yaml [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+  - /lib/cmake
+  - /lib/pkgconfig
+
+modules:
+  - name: pinentry
+    buildsystem: autotools
+    config-opts:
+      - --enable-maintainer-mode
+      - --enable-pinentry-qt
+    cleanup:
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:30]: pinentry shouldn't be cleaned up, I said it for boost

File: com.bktus.gpgfrontend.yaml (hunk 6 above)
Comment (bbhtt) [reply com.bktus.gpgfrontend.yaml:44]: Drop this builder can build this by default.

File: com.bktus.gpgfrontend.yaml [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+  - '*.la'
+  - /share/doc
+  - /share/gtk-doc
+  - /share/man
+  - /include
+  - /lib/cmake
+  - /lib/pkgconfig
+
+modules:
+  - name: pinentry
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:25]: You removed the sources part from pinentry too

File: com.bktus.gpgfrontend.yaml [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+command: GpgFrontend
+
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+  - --filesystem=home
+  - --filesystem=/run/user
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:14]: Is this for gpg agent socket? Then what you want is `--socket=gpg-agent` which gives access to `gpgconf --list-dirs agent-socket`

File: com.bktus.gpgfrontend.yaml [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+sdk: org.kde.Sdk
+command: GpgFrontend
+
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+  - --filesystem=home
```
Comment (bbhtt) [inline com.bktus.gpgfrontend.yaml:13]: Is this for `~/.gnupg`, you can restrict it to that only

**Comment by bbhtt** [comment]
bot, build com.bktus.gpgfrontend

File: com.bktus.gpgfrontend.yaml [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+  - /include
+  - /lib/cmake
+  - /lib/pkgconfig
+
+modules:
+  - name: boost
+    buildsystem: simple
+    build-commands:
+      - ./bootstrap.sh --prefix=${FLATPAK_DEST} --with-libraries=date_time,system
+      - ./b2 variant=release link=static runtime-link=shared cxxflags="$CXXFLAGS" linkflags="$LDFLAGS" -j $(nproc) install
```
Comment (hfiguiere) [inline com.bktus.gpgfrontend.yaml:31]:
```suggestion
      - ./b2 variant=release link=static runtime-link=shared cxxflags="$CXXFLAGS" linkflags="$LDFLAGS" -j $FLATPAK_BUILDER_N_JOBS install
```

**Comment by hfiguiere** [comment]
```
ninja: error: '/usr/lib/x86_64-linux-gnu/libicui18n.so', needed by 'src/core/libgpgfrontend_core.so', missing and no known rule to make it
```

**Comment by saturneric** [comment]
> ```
> ninja: error: '/usr/lib/x86_64-linux-gnu/libicui18n.so', needed by 'src/core/libgpgfrontend_core.so', missing and no known rule to make it
> ```

Solved it. Hard coded icu library links found at project configuration files.

**Comment by bbhtt** [comment]
bot, build com.bktus.gpgfrontend

**Pull request closed without merging**